console/console-init/ui/src/modules/device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from "react";
+import {
+  Split,
+  SplitItem,
+  Title,
+  Flex,
+  FlexItem,
+  DropdownItem,
+  DropdownPosition,
+  PageSection,
+  PageSectionVariants,
+  DropdownSeparator
+} from "@patternfly/react-core";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { FormatDistance } from "use-patternfly";
+import { DropdownWithKebabToggle, SwitchWithToggle } from "components";
+
+interface IDeviceDetailHeaderProps {
+  deviceName: string;
+  addedDate: string;
+  lastTimeSeen: string;
+  onEdit: () => void;
+  onChange: (enabled: boolean) => void;
+  onDelete: () => void;
+  onClone: () => void;
+}
+
+export const project_header_styles = StyleSheet.create({
+  flex_left_border: {
+    paddingLeft: "1em",
+    borderLeft: "0.05em solid",
+    borderRightColor: "--pf-chart-global--Fill--Color--200"
+  },
+  kebab_toggle_margin: {
+    marginTop: 20,
+    marginLeft: 10,
+    fontSize: 15
+  },
+  namespace_info_margin: {
+    marginTop: 16,
+    marginBottom: 24
+  },
+  no_bottom_padding: {
+    paddingBottom: 0
+  }
+});
+
+const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
+  deviceName,
+  addedDate,
+  lastTimeSeen,
+  onEdit,
+  onDelete,
+  onChange,
+  onClone
+}) => {
+  const DeviceTitle = () => (
+    <Split gutter="md">
+      <SplitItem>
+        <Title headingLevel="h1" size="4xl" id="device-name">
+          {deviceName}
+        </Title>
+      </SplitItem>
+    </Split>
+  );
+
+  const DeviceDetailInFlex = () => (
```

Comment:
I don't think so we need Flex postfix in function name. can we try something like DeviceDateSection

Reply:
Like my earlier comment, as per my POV, the segregated code looks cleaner. :)